lib/cli/index.js

```diff
@@ -0,0 +1,241 @@
+    }
+];
+
+function CLI() {
+    Cmdln.call(this, {
+        name: 'kbmctl',
+        desc: packageJson.description,
+        options: OPTIONS,
+        helpOpts: {
+            includeEnv: true,
+            minHelpCol: 30
+        },
+        helpSubcmds: [
+            'help',
+            'recovery',
+            'pivtoken'
+        ],
+        helpBody: [
+            /* eslint-disable max-len */
+            'Exit Status:',
+            '    0   Successful completion.',
+            '    1   An error occurred.',
+            '    2   Usage error.',
+            '    3   "ResourceNotFound" error (when a config, pivtoken, etc. with',
+            '        the given name or id is not found) or "PIVTokenDeleted" error.'
+            /* eslint-enable max-len */
+        ].join('\n')
+    });
+}
+util.inherits(CLI, Cmdln);
+
+CLI.prototype.init = function (opts, args, callback) {
+    var self = this;
+    this.opts = opts;
+
+    this.log = bunyan.createLogger({
+        name: this.name,
+        serializers: bunyan.stdSerializers,
+        stream: process.stderr,
+        level: 'warn'
+    });
+
+    if (opts.verbose) {
+        this.log.level('trace');
+        this.log.src = true;
+        this.showErrStack = true;
+    }
+
+    if (opts.version) {
+        console.log('KBMAPI CTL', packageJson.version);
+        console.log(packageJson.homepage);
+        callback(false);
+        return;
+    }
+
+    this.__defineGetter__('kbmapi', function getTritonapi() {
+        if (self._kbmapi === undefined) {
+            var host = 'localhost';
+            var port = process.env.KBMAPI_PORT || 80;
+
+            if (opts.host) {
+                host = opts.host;
+            } else if (process.env.KBMAPI_HOST) {
+                host = process.env.KBMAPI_HOST;
+            } else {
+                port = config.port;
+            }
+
+
+            var reqID = UUID.v4();
+            var clientOpts = {
+                agent: false,
+                headers: { 'x-request-id': reqID },
+                url: format('http://%s:%d', host, port)
+            };
+
+            self._kbmapi = new kbmapi(clientOpts);
+            self._kbmapi.req_id = reqID;
+            self.log.trace('created kbmapi');
+        }
+        return self._kbmapi;
+    });
+
+    // Cmdln class handles `opts.help`.
+    Cmdln.prototype.init.call(self, opts, args, callback);
+};
+
+CLI.prototype.fini = function fini(subcmd, err, cb) {
+    this.log.trace({err: err, subcmd: subcmd}, 'cli fini');
+    if (this._kbmapi) {
+        delete this._kbmapi;
+    }
+    cb();
+};
+
+CLI.prototype.do_recovery = require('./do_recovery');
+CLI.prototype.do_pivtoken = require('./do_pivtoken');
+CLI.prototype.do_completion = require('./do_completion');
+// --- mainline
+
+function main(argv) {
+    if (!argv) {
+        argv = process.argv;
+    }
+
+    var cli = new CLI();
+    cli.main(argv, function (err) {
+        var exitStatus = (err ? err.exitStatus || 1 : 0);
```

Comment:
2 and 3 are used by node-triton and kept here just in case we want to use them the same way we use them there. In order to set it the error needs to set the value when appropriate.

Actually, node-triton just uses `3` exit status for instance not found or image not found errors. So far, those are not used by kbmctl for now.